response.go

```diff
@@ -2,23 +2,68 @@ package tarantool
 
 import (
 	"fmt"
+	"io"
+	"io/ioutil"
 
 	"github.com/tarantool/go-iproto"
 	"github.com/vmihailenco/msgpack/v5"
 )
 
-type Response struct {
-	RequestId uint32
-	Code      uint32
-	// Error contains an error message.
-	Error string
-	// Data contains deserialized data for untyped requests.
-	Data []interface{}
-	// Pos contains a position descriptor of last selected tuple.
-	Pos      []byte
-	MetaData []ColumnMetaData
-	SQLInfo  SQLInfo
-	buf      smallBuf
+// Response is an interface with operations for the server responses.
+type Response interface {
+	// Header returns a response header.
+	Header() Header
+	// Decode decodes a response.
+	Decode() ([]interface{}, error)
+	// DecodeTyped decodes a response into a given container res.
+	DecodeTyped(res interface{}) error
+}
+
+type baseResponse struct {
+	// header is a response header.
+	header Header
+	// data contains deserialized data for untyped requests.
+	data []interface{}
+	buf  smallBuf
+	// Was the Decode() func called for this response.
+	decoded bool
+	// Was the DecodeTyped() func called for this response.
+	decodedTyped bool
+	err          error
+}
+
+func createBaseResponse(header Header, body io.Reader) (baseResponse, error) {
+	if body == nil {
+		return baseResponse{header: header}, nil
+	}
+	if buf, ok := body.(*smallBuf); ok {
+		return baseResponse{header: header, buf: *buf}, nil
+	}
+	data, err := ioutil.ReadAll(body)
+	if err != nil {
+		return baseResponse{}, err
+	}
+	return baseResponse{header: header, buf: smallBuf{b: data}}, nil
+}
+
+// SelectResponse is used for the select requests.
+// It might contain a position descriptor of the last selected tuple.
+type SelectResponse struct {
+	baseResponse
+	// pos contains a position descriptor of last selected tuple.
+	pos []byte
+}
+
+// PrepareResponse is used for the prepare requests.
+// It might contain meta-data and sql info.
+type PrepareResponse ExecuteResponse
```

Comment:
Or something like this.
```suggestion
// PrepareResponse is used for the prepare requests.
// It might contain meta-data and sql info.
//
// Be careful: now this is an alias for `ExecuteResponse`, but it could be change in the future.
You need to cast to `PrepareResponse` a response from `PrepareRequest`.
type PrepareResponse ExecuteResponse
```